pkg/metrics/metrics.go

```diff
@@ -582,12 +584,31 @@ func prometheusToken(c client.Client) (*string, error) {
 	}
 
 	if len(token) == 0 {
-		return nil, fmt.Errorf("failed to find token secret for prometheus-k8s SA")
+		// If serviceaccount token doesn't exist, try to use token request to get a token
+		token, err = RequestPrometheusServiceAccountAPIToken(c)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find token secret for prometheus-k8s SA due to %v", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to request token for prometheus-k8s SA due to %v", err)
```

Reply:
I used get here because here also means the read legacy-token failed to.